packages/useful-types/CHANGELOG.md

```diff
@@ -7,11 +7,20 @@ and adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 <!-- ## Unreleased -->
 
+### Breaking Change
+
+- `Omit` has been removed. You should be able to remove any `import { Omit } from '@shopify/useful-types';` and it should fallback on `Omit` from Typescript directly
+- `ThenType` has been removed. You should use the native `Awaited` type from Typescript instead
+- `Arguments` has been removed. You should replace any of its usage in your project with the native `Parameters` from Typescript. On the same note, replace `ArgumentAtIndex` by `Parameters<T>[i]` and `FirstArgument<T>` by `Parameters<T>[0]`.
+- `ConstructorArguments` has been removed. You should replace any of its usage in your project with the native `ConstructorParameters` from Typescript. On the same note, replace `ConstructorArgumentAtIndex` by `ConstructorParameters<T>[i]` and `FirstConstructorArgument<T>` by `ConstructorParameters<T>[0]`.
+- `MaybeFunctionReturnType` has been removed. Each usage should be replaced by the native `ReturnType` of Typescript
+
 ## 3.1.0 - 2022-02-09
 
 ### Added
 
 - Added `PartialSome` and `RequireSome` types to set specified fields of a property to optional or required
+- Deprecate `ThenType`, `Omit`, `Arguments`, `ArgumentAtIndex`, `FirstArgument`, `MaybeFunctionReturnType`, `ConstructorArguments`, `ConstructorArgumentAtIndex` and `FirstConstructorArgument` in favour of their Typescript counterparts
```

Comment:
I don't think we need this line in the changelog of 3.1.0 - we never marked these types with `@deprecated` or any other mechanism. I reckon we should remove it.